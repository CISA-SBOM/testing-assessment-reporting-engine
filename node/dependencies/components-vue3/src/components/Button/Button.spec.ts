import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from './Button.vue'

describe('Button', () => {
  it('should match its default snapshot', () => {
    const wrapper = shallowMount(Component, {
      slots: {
        default: 'Button'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should match its variant primary snapshot', () => {
    const wrapper = shallowMount(Component, {
      slots: {
        default: 'Button'
      },
      props: {
        variant: 'primary'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})