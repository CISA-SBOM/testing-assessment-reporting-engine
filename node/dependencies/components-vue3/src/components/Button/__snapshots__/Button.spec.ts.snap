// Vitest Snapshot v1

exports[`Button > should match its default snapshot 1`] = `
<button
  class=""
  data-id="sds-button"
>
  
  Button
  
</button>
`;

exports[`Button > should match its variant primary snapshot 1`] = `
<button
  class="btn btn-primary"
  data-id="sds-button"
>
  
  Button
  
</button>
`;
